src/ts/components/settings/SettingsPanel.ts

```diff
@@ -60,7 +60,7 @@ export enum NavigationDirection {
  *
  * @category Components
  */
-export class SettingsPanel extends Container<SettingsPanelConfig> {
+export class SettingsPanel<Config extends SettingsPanelConfig> extends Container<Config> {
```

Comment:
nit: I feel like `Config` does not indicate that the type is generic. I think convention would be something like `TConfig`, or a single character, like `C`. But it should also fit with the generic naming of the `Container` class, so feel free to ignore

Reply:
Agree but it's arealdy called `Config` everywhere in the code base.